x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java

```diff
@@ -280,12 +281,18 @@ protected void nodeOperation(AllocatedPersistentTask task, @Nullable TransformTa
 
         // <3> Validate the transform, assigning it to the indexer, and get the field mappings
         ActionListener<TransformConfig> getTransformConfigListener = ActionListener.wrap(config -> {
-            if (config.isValid()) {
+            ValidationException validationException = config.validate(null);
+            if (validationException == null) {
                 indexerBuilder.setTransformConfig(config);
                 SchemaUtil.getDestinationFieldMappings(buildTask.getParentTaskClient(), config.getDestination().getIndex(),
                         getFieldMappingsListener);
             } else {
-                markAsFailed(buildTask, TransformMessages.getMessage(TransformMessages.TRANSFORM_CONFIGURATION_INVALID, transformId));
+                markAsFailed(
```

Comment:
if I remember correctly `markAsFailed` does not audit. I think we should add an audit here, so users will see the error in the job messages.

Reply:
you are right, this looks inconsistent, but its already inconsistent today. The last step audits, too. The in-between ones don't audit but only log.
I think we should audit in all failure cases, logs might not be accessible to the user.